hail/python/hailtop/batch/contrib/regenie/regenie-batch.py

```diff
@@ -0,0 +1,293 @@
+    if ninputs == 0:
+        raise Exception("An input file is required. Please specify: --bed or --bgen or --pgen")
+
+    if ninputs > 1:
+        raise Exception(f"""More than one input file specified.
+                            Check that only one of --bed or --bgen or --pgen are specified for --step {r.step}
+                        """)
+
+    if r.step == 2:
+        if r.pred:
+            print("--pred provided for --step 2, but Batch will constrain the --pred output prefix to that of --step 1.")
+
+        if r.extract or r.exclude:
+            raise Exception("--extract and --exclude only work with --step 1")
+
+    if r.lowmem and not r.lowmem_prefix:
+        raise Exception("When --lowmem provided, --lowmem-prefix required")
+
+    return r
+
+
+# TODO: check --phenoCol, --phenoColList, --remove
+def get_phenos(step_args: argparse.Namespace):
+    with hopen(step_args.phenoFile, "r") as f:
+        return f.readline().strip().split(" ")[2:]
+
+
+def get_input(batch, step_args: argparse.Namespace):
+    add = {}
+    for name, val in vars(step_args).items():
+        if name in from_underscore:
+            name = from_underscore[name]
+
+        if name not in input_file_args or val is None:
+            continue
+
+        if name == "bed":
+            prefix = step_args.bed
+            inf = {}
+            inf["bed"] = f"{prefix}.bed"
+            inf["bim"] = f'{prefix}.bim'
+            inf["fam"] = f'{prefix}.fam'
+            add[name] = batch.read_input_group(**inf)
+            continue
+
+        if name == "pgen":
+            prefix = step_args.pgen
+            inf = {}
+            inf["pgen"] = f'{prefix}.pgen'
+            inf["pvar"] = f'{prefix}.pvar'
+            inf["psam"] = f'{prefix}.psam'
+            add[name] = batch.read_input_group(**inf)
+            continue
+
+        add[name] = batch.read_input(val)
+
+    return add
+
+
+def prepare_jobs(batch, args: BatchArgs, step1_args: argparse.Namespace,
+                 step2_args: argparse.Namespace):
+    j1 = batch.new_job(name='run-regenie')
+    j1.image('akotlar/regenie:latest')
+    j1.cpu(args.cores)
+    j1.memory(args.memory)
+    j1.storage(args.storage)
+
+    in_step1 = get_input(batch, step1_args)
+
+    phenos = get_phenos(step1_args)
+    nphenos = len(phenos)
+
+    s1pre = step1_args.out
+    s1out = {"log": f"{s1pre}.log", "pred_list": f"{s1pre}_pred.list"}
+
+    for i in range(1, nphenos + 1):
+        s1out[f"{s1pre}_{i}"] = f"{s1pre}_{i}.loco"
+
+    if step1_args.lowmem:
+        for i in range(1, nphenos + 1):
+            pfile = f"{step1_args.lowmem_prefix}_l0_Y{i}"
+            s1out[pfile] = pfile
+
+    j1.declare_resource_group(**{s1pre: s1out})
+
+    cmd1 = ""
+    for name, val in vars(step1_args).items():
+        if val is None or val is False:
+            continue
+
+        if name in from_underscore:
+            name = from_underscore[name]
+
+        # pred is not used in step 1 according to documentation
```

Comment:
Sounds like we should remove it from the argument parser for step1, yeah? I prefer errors to silently dropping things, when possible.

Reply:
Having to verify every valid regenie argument combination seems wrong. We're providing a way to run regenie, so I'd like to keep it as open as possible and let regenie complain. I agree with your earlier arguments, wrt lowmem.